rewriter/ClassNew.cc

```diff
@@ -12,16 +12,12 @@ namespace sorbet::rewriter {
 
 // Is this expression a synthetic `T.bind` call we added from the `ClassNew` rewriter on sends?
 bool isRewrittenBind(ast::ExpressionPtr &expr) {
-    auto send = ast::cast_tree<ast::Send>(expr);
-    if (send == nullptr) {
```

Comment:
I'm slightly worried about these sorts of changes to `Rewriter`, because it makes individual rewriter passes sensitive to whether they've run before or after the `TypeAssertion` rewriter. This sort of thing has caused bugs in the past, as people handle one case but forget to handle another case.

I guess you've chosen to do it here instead of in `Namer::findSymbols` to avoid having to make `findSymbols` be a non-shallow map?

Another alternative might be to lurk this in `Cleanup.cc` so that it can run after all the rewriter passes.

Reply:
Another alternative would be to do this in `Desugar.cc`, so that *all* the rewriter passes can assume that it will already have run, which should allow rewriter passes to never have to do their own `T.let` parsing, regardless of the order of rewriter passes.

Reply:
I can believe that ordering issues have caused problems in the past, but I'm not imaginative enough to see why this particular change would cause problems: the rewriter here is guaranteed to be run before all of the rewriters on class bodies, which means the latter will always see `Cast` nodes.  It is really nice that the class body rewriters can depend on this, because then they don't need custom parsing logic (cf. the minitest changes).

All of the other `Send`-based rewriters are more or less insensitive to the ordering issues you mention?  (I could see problems happening if some things were `preTransform*`, so perhaps we could have future problems...)  Even if other `Send`-based rewriters are examining their arguments, those arguments have already been run through the cast rewriter.

Moving things into desugar could be a reasonable thing to do, though I'm not excited about adding even more logic to desugaring sends.

Reply:
Sounds good, we can leave it here for now and if we need to move it in the future we can